File: misanuncios-ui/src/app/infrastructure/dto/userHasRole.dto.ts
import { RoleDto } from "./role.dto";
import { UserDto } from "./user.dto";

import { UserHasRoleKeyDto } from "./userHasRoleKey.dto";

export interface UserHasRoleDto
{
  id: UserHasRoleKeyDto;
  role: RoleDto; // Assuming Role model exists
  user: UserDto; // Assuming User model exists

}

File: misanuncios/src/main/java/com/yaagoub/misanuncios/infrastructure/rest/spring/dto/UserHasRoleDto.java
package com.yaagoub.misanuncios.infrastructure.rest.spring.dto;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@Data
@EqualsAndHashCode(exclude = { "id","role","user"})
public class UserHasRoleDto {
    private UserHasRoleKeyDto id;

    private RoleDto role;


    private UserDto user;
}
